pkg/eventqueue/eventqueue.go

```diff
@@ -169,11 +176,16 @@ func (ev *Event) WasCancelled() bool {
 // closed, indicating that the Event was not ran. This function may block if
 // the queue is at its capacity for events.
 func (q *EventQueue) Enqueue(ev *Event) <-chan interface{} {
-
 	if q.notSafeToAccess() || ev == nil {
 		return nil
 	}
 
+	// Events can only be enqueued once.
+	if val := ev.enqueued.Load(); val == true {
+		return nil
+	}
+	ev.enqueued.Store(true)
```

Comment:
This does not prevent two concurrent Enqueue()s for the same event, if they both Load `enqueued` before them both storing `true`. Maybe `if atomic.AddUint32(&ev.enqueued, 1) > 1 { return nil }` could be used instead?

Reply:
You are correct, done!